hack/builder/Dockerfile

```diff
@@ -112,17 +75,12 @@ RUN \
     go install && \
     cd /test-infra/robots/issue-creator && \
     go install && \
-    cd /test-infra/robots/pr-labeler && \
-    go install && \
-    cd /go && \
-    go clean -cache -modcache -r && \
-    rm -rf /test-infra && \
-    rm -rf /go && mkdir /go
```

Comment:
why are these lines removed? this should help shrink the image size

Reply:
We should keep that but withou -r flag and `cd /go`